### PR TITLE
feat: User partial implement

### PR DESCRIPTION
**Pull request by hsuanyu414**
1. User model
2. UserCreate's repo, domain, service, representation

### CODE REVIEW

File: backend/src/presentations/UserPresentation.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
+package presentations
+
+import (
+	// "fmt"
```
Comment (ra890927) [inline backend/src/presentations/UserPresentation.go:4]: delete

File: backend/src/domains/UserDomain.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+type UserDomain interface {
+	CreateUser(user *models.User) (*models.User, error)
+	UpdateUser(user *models.User) (*models.User, error)
+}
+
+type userDomain struct {
+	UserRepo repos.UserRepo
+}
+
+func NewUserDomain(opt ...repos.UserRepo) UserDomain {
```
Comment (ra890927) [inline backend/src/domains/UserDomain.go:17]: inspect `len(opt) > 1`

File: backend/src/domains/UserDomain.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	if len(opt) == 1 {
+		return &userDomain{
```
Comment (ra890927) [inline backend/src/domains/UserDomain.go:19]: return struct is not match func defined

File: backend/src/domains/UserDomain.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	"meeting-center/src/repos"
+)
+
+type UserDomain interface {
+	CreateUser(user *models.User) (*models.User, error)
+	UpdateUser(user *models.User) (*models.User, error)
+}
+
+type userDomain struct {
+	UserRepo repos.UserRepo
```
Comment (ra890927) [inline backend/src/domains/UserDomain.go:14]: `UserRepo` 不必定義 public

File: backend/src/domains/UserDomain.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+			UserRepo: opt[0],
+		}
+	} else {
+		return &userDomain{
+			UserRepo: repos.NewUserRepo(),
+		}
+	}
+}
+
+func (ud *userDomain) CreateUser(user *models.User) (*models.User, error) {
```
Comment (ra890927) [inline backend/src/domains/UserDomain.go:29]: receiver 不用 pointer

File: backend/src/presentations/UserPresentation.go (hunk 1 above)
Comment (ra890927) [inline backend/src/presentations/UserPresentation.go:4]: delete

File: backend/src/repos/UserRepo.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+)
+
+type UserRepo interface {
+	CreateUser(user *models.User) (*models.User, error)
+
+	UpdateUser(user *models.User) (*models.User, error)
+}
+
+type userRepo struct {
+	dsn string
```
Comment (ra890927) [inline backend/src/repos/UserRepo.go:18]: dsn 是什麼意思

Comment (hsuanyu414) [reply backend/src/repos/UserRepo.go:18]: data source name

**Review by ra890927 — APPROVED** [review]

**Pull request merged**